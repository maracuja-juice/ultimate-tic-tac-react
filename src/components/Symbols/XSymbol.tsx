import * as React from 'react';
import classNames from 'classnames';
import styles from './Symbols.module.css';
import { XGameIcon } from '../Icons';

export interface XSymbolProps {
  shouldAnimate?: boolean;
}

export class XSymbol extends React.Component<XSymbolProps> {
  public static defaultProps: Partial<XSymbolProps> = {
    shouldAnimate: true,
  };

  render() {
    const { shouldAnimate } = this.props;
    const iconClass = classNames({
      [styles.x]: true,
      [styles.animateX]: shouldAnimate,
    });

    return <XGameIcon className={iconClass} />;
  }
}
