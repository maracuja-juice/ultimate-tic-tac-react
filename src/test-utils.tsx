// See: https://testing-library.com/docs/react-testing-library/setup#custom-render
import React, { FC, ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { IconProvider } from './components/Icons';
import '@testing-library/jest-dom'; // https://github.com/testing-library/jest-dom/issues/282#issuecomment-660034778
import { MemoryRouter } from 'react-router-dom';

const AllTheProviders: FC = ({ children }) => {
  return (
    <IconProvider>
      <MemoryRouter>{children}</MemoryRouter>
    </IconProvider>
  );
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';

export { customRender as render };
