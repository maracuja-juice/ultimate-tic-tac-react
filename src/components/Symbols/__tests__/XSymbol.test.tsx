import * as React from 'react';
import { render } from '../../../test-utils';
import { XSymbol } from '../XSymbol';

test('renders x', () => {
  // This is a hack and should not be used in tests where we can queryByRole or other!
  const { container } = render(<XSymbol />);
  const x = container.querySelector('svg');
  expect(x).not.toBeNull();
  expect(x!.classList.value).toMatch(/animatex/i);
});

test('renders x animated', () => {
  // This is a hack and should not be used in tests where we can queryByRole or other!
  const { container } = render(<XSymbol shouldAnimate={true} />);
  const x = container.querySelector('svg');
  expect(x).not.toBeNull();
  expect(x!.classList.value).toMatch(/animatex/i);
});

test('renders x not animated', () => {
  // This is a hack and should not be used in tests where we can queryByRole or other!
  const { container } = render(<XSymbol shouldAnimate={false} />);
  const x = container.querySelector('svg');
  expect(x).not.toBeNull();
  expect(x!.classList.value).not.toMatch(/animatex/i);
});
