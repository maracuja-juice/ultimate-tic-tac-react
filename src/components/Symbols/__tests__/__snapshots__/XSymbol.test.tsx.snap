// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`XSymbol should match snapshot 1`] = `
<ForwardRef(SvgX)
  aria-label="X"
  className="x animateX"
/>
`;

exports[`XSymbol should match snapshot 2`] = `
<ForwardRef(SvgX)
  aria-label="X"
  className="x"
/>
`;
